src/library/autodj/dlgautodj.ui

```diff
@@ -130,11 +130,14 @@
      <item>
       <widget class="QPushButton" name="pushButtonAddRandom">
        <property name="toolTip">
-        <string>Adds a random track from track sources (crates) to the Auto DJ queue.
-If no track sources are configured, the track is added from the library instead.</string>
+        <string>Inserts on top of que a realated track.
+  If Auto DJs crates contains titles, add random from crates
+  If CTRL is pressed, add related to the playing track
+  If any track is selected, add realated to the selected
+  If no track is selected, add related to the last track in que</string>
```

Comment:
Is there really a use case for picking any similar title from the whole library if the user has already assigned a number of tracks where Mixxx should pick from.  
Did you think about always picking a similar artist from the track source the original random feature uses? 

How about that: 

* Add a Auto DJ preference option: 
[  ] Improve random title using similar artist from last.fm 

An issue I can imagine with this is that with a small track pool we more likely get the same order on a second run.

Reply:
1)
Please read the quoted IFs as a sequence; the first IF-rule that hits will do the job. E.g. IF thre are tracks in the AutoDJ crate, the crate will used and end of story.

2)
No the order is unlikely the same, that is the clever hack! Look in the code for the comment `"if we always take the best, it's boring"`

Reply:
Ah, got it. 
Is my use case supported? Let's say I am doing a 90s party, I add a crate to AutoDJ containig all my 90s Tracks. This feature could then pick similar artists, but only tracks from the 90s.

Reply:
At the moment, no. But is is very simple to change in that way. The AutoDJ crate has to old behavior (nearly full random), which I not removed.

Here ist the place where it happens:
 ```
           if (m_crateList.isEmpty()) {
                // Fetch Track from Library since we have no assigned crates
                randomTrackId = m_autoDjCratesDao.getRandomTrackIdFromLibrary(
                        m_iAutoDJPlaylistId, sArtist);
            } else {
                // Fetch track from crates.
                // We do not fall back to Library if this fails because this
                // may add banned tracks
                randomTrackId = m_autoDjCratesDao.getRandomTrackId();
            }

```